shared/src/main/scala/effekt/Typer.scala

```diff
@@ -557,7 +559,7 @@ class Typer extends Phase[Module, Module] { typer =>
     //   BlockArg: foo { n => println("hello" + n) }
     //     or
     //   BlockArg: foo { (n: Int) => println("hello" + n) }
-    def checkBlockArgument(tpe: BlockType, arg: source.BlockArg): Unit = Context.at(arg) {
+    def checkBlockArgument(tpe: BlockType, arg: source.BlockArg, isResume: Boolean): Unit = Context.at(arg) {
```

Comment:
I would revert the changes to `checkBlockArgument`. Instead for now it would suffice to reject `resume(v)` for an bidirectional operation and `resume { e }` for a non-bidirectional one. It would be great to have an error in that case that explains the situation well.